iib/workers/tasks/opm_operations.py

```diff
@@ -184,60 +187,78 @@ def get_opm_port_stacks() -> Tuple[List[List[int]], List[str]]:
     return ports_list, port_purposes
 
 
-def create_port_filelocks(func: Callable) -> Callable:
+def create_port_filelocks(port_purposes: List[str]) -> Callable:
     """
     Create a file-lock on random port from the configured range.
 
-    :param function func: the function to be decorated
-    :rtype: function
-    :return: the decorated function
+    :param List[str] port_purposes: the list of port purposes to be locked
+    :rtype: Callable
+    :return: the decorator function
     """
 
-    @wraps(func)
-    def inner(*args, **kwargs):
+    def decorator(func: Callable) -> Callable:
+        """
+        Create a file-lock on random port from the configured range.
 
-        port_stacks, port_purposes = get_opm_port_stacks()
-        # Attempt to acquire the lock for each port in the range (shuffled order)
-        lock_success = False
+        :param function func: the function to be decorated
+        :rtype: function
+        :return: the decorated function
+        """
 
-        # Initialize the generator
-        gen = port_file_locks_generator(port_stacks=port_stacks, port_purposes=port_purposes)
+        @wraps(func)
+        def inner(*args, **kwargs):
 
-        # Use the function to retrieve values from the generator
-        while not lock_success:
-            new_locks = next(gen)
-            currently_active_locks = []
+            # If we do not have any ports to lock
+            if len(port_purposes) == 0:
+                return func(*args, **kwargs)
```

Comment:
This part is new. The rest of the create_port_filelocks is just shifted by one Tab.

Reply:
The reason for the introduction of this part is the `generate_cache_locally` method, which needs only `opm_pprof_port`, but only with opm version 1.40.0, so in case of opm version 1.26.4, we need to be able to bypass the `port_file_locks_generator` (which would throw an error, as it would get empty port_stack).